### ISSUE TITLE
blogs(n) greater than num_recent_posts

### ISSUE THREAD

**Issue by richp10** [alanorth/hugo-theme-bootstrap4-blog#75]
Great theme - but I have one problem. 

It seems to me that if you have more blog posts than num_recent_posts there is no way to access older posts other than by categories.  Most themes provide a sidebar listing of years so you can drill down. 

Am I missing something / are you planning anything like this. Thanks again for the template..

**Comment by fte378** [comment]
I added this myself. For an example see  https://frank.tegtmeyer.net/

The additions are an archive list sorted by year and a previous/next navigation in side the posts.

If that's what you want I can give you my changes.

**Comment by alanorth** [comment]
Thanks @richp10. If I understand correctly you mean accessing older posts from the sidebar itself? When I made this theme it was basically just a spiritual copy of the [Bootstrap Blog theme](https://getbootstrap.com/docs/4.0/examples/blog) so I didn't think about much beyond that. I think date-based archives like in WordPress themes could be a nice addition, though. Can you point me to some Hugo themes which do that?

**Comment by Jieiku** [comment]
I have an additional thought about this... lets say your on page 6 of the blog, you could have something like this:

[<][1]...[4][5][6][7][8]...[29][>]

![image](https://user-images.githubusercontent.com/106644/73040129-4452de80-3e0d-11ea-8757-1b696ec351ca.png)

![image](https://user-images.githubusercontent.com/106644/73040145-50d73700-3e0d-11ea-8ea4-a1aabd9aed46.png)

The above screenshots are from a wordpress site.
I will have to see if I can find a similar page selector for Hugo.

We do have the previous and next page buttons, so maybe that is sufficient.
I have also thought about adding a drop down menu to go through archives.
I am still thinking about the best solution for browsing old posts.

**Comment by alanorth** [comment]
@xekon yes this is nice, but my goal with the theme was to replicate the traditional [Bootstrap blog style as it appeared in version 4 alpha](https://v4-alpha.getbootstrap.com/examples/blog/).

**Comment by Jieiku** [comment]
I am happy that you did! When I chose to use this theme, I evaluated all the themes listed here: https://themes.gohugo.io/  My requirements were low requests per page, and low file size transferred. Bootstrap is the most used theme for Drupal, so when I came across this theme of yours I was pretty much sold, there were a few other bootstrap themes, but they did not perform as well as this one.

I actually decided against adding a page selector like that. Instead I plan to leave it how it is and just add a search function. I plan to use https://cse.google.com/cse/ as discussed here: https://github.com/alanorth/hugo-theme-bootstrap4-blog/issues/66

With a search function on the site, there is less need to provide a way to sift through old posts, instead they can just search for whatever it is they are trying to find.

**Comment by richp10** [comment]
An old thread (thanks for bumping xekon) - looking back, I like what @fte378 did and wonder if you could share the code to achieve this.  To be honest, your site is *exactly* how I would like to use this theme.

**Comment by Jieiku** [comment]
@richp10 ah yes they have an overviews section to list all posts, categories and tags:

![image](https://user-images.githubusercontent.com/106644/73051242-239f7e80-3e37-11ea-833b-6e308aaaf853.png)

That does look like a really clean solution! Could you please share @fte378 I am interested in the code as well.

**Comment by fte378** [comment]
@xekon, @richp10:  Attached you can find my overrides for the theme. I don't know if it's elegant or the 'correct' Hugo way - I only look at it when it starts to fail and try to fix it then.

When I remember correctly
 the main part is layouts/archivetype/single.html

[layouts.zip](https://github.com/alanorth/hugo-theme-bootstrap4-blog/files/4114068/layouts.zip)

**Comment by Jieiku** [comment]
Thank you @fte378 be sure to checkout: https://github.com/alanorth/hugo-theme-bootstrap4-blog/pull/127

It solves the issue of loading Fontawesome that you were wanting to resolve.

**Comment by Jieiku** [comment]
@fte378 I was able to implement this on my site, thank you. I had to pay close attention to the path of my files, IE post vs posts and rename layouts\archivetype  to layouts\archives
I also made an /content/archives.md:
```
---
title: "Archives"
type: archives
url: "/archives/"
---
```
One more item off my todo list, thanks :)

I will add the required changes to a clean branch in my repository and submit as a pull request.

**Comment by alanorth** [comment]
@fte378 thank you for sharing your solution. I will take a look at it, and search around for what [other people](https://randomgeekery.org/2016/04/17/yearly-post-archives-in-hugo/) are doing, or perhaps look into [Hugo's list templates](https://gohugo.io/templates/lists/#by-date-1) and grouping by date.

**Comment by Jieiku** [comment]
@alanorth are you planning to include a feature like this?

If so I wont bother with putting together a pull request.

here it is implemented on my site: https://www.solosaga.com/
